src/content/getusermedia/record/js/main.js

```diff
@@ -138,20 +117,31 @@ function startRecording() {
 function stopRecording() {
   mediaRecorder.stop();
   console.log('Recorded Blobs: ', recordedBlobs);
-  recordedVideo.controls = true;
 }
 
 function handleSuccess(stream) {
   recordButton.disabled = false;
-  console.log('getUserMedia() got stream: ', stream);
+  console.log('getUserMedia() got stream:', stream);
   window.stream = stream;
 
   const gumVideo = document.querySelector('video#gum');
   gumVideo.srcObject = stream;
 }
 
-function handleError(error) {
-  console.log('navigator.getUserMedia error: ', error);
+async function init() {
+  try {
+    const stream = await navigator.mediaDevices.getUserMedia(constraints);
+    handleSuccess(stream);
+  } catch (e) {
+    console.error('navigator.getUserMedia error:', e);
+  }
 }
 
-navigator.mediaDevices.getUserMedia(constraints).then(handleSuccess).catch(handleError);
+// window.isSecureContext could be used for Chrome
+let isSecureOrigin = location.protocol === 'https:' || location.hostname === 'localhost';
```

Comment:
we are requiring https on github pages so I don't think we need to handle this error

Reply:
👍